### PR TITLE
Bugfix welcome screen : add missing close button

### PR DESCRIPTION
**Pull request by mejdoo**
### Issues

There is a missing close button that should appear on the new welcome screen when we on click on "New Team or Account" from profile screen.
#### APK
[Download build #1090](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1090/artifact/build/artifact/wire-dev-PR2579-1090.apk)
[Download build #1091](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1091/artifact/build/artifact/wire-dev-PR2579-1091.apk)

### CODE REVIEW

File: app/src/main/scala/com/waz/zclient/LaunchActivity.scala [hunk 1]
```diff
@@ -96,7 +96,7 @@ class LaunchActivity extends AppCompatActivity with ActivityHelper with DerivedL
   }
 
   private def startSignUp() = {
-    startActivity(AuthenticationActivity.newIntent(this))
+    startActivity(AuthenticationActivity.newIntent(this,false))
```
Comment (BradleyWire) [inline app/src/main/scala/com/waz/zclient/LaunchActivity.scala]: formatting

File: app/src/main/scala/com/waz/zclient/preferences/pages/ProfileView.scala [hunk 2]
```diff
@@ -99,8 +99,7 @@ class ProfileViewImpl(context: Context, attrs: AttributeSet, style: Int) extends
     newTeamButton.setVisible(true)
     newTeamButton.onClickEvent.on(Threading.Ui) { _ =>
       // We want to go directly to the landing page.
-      val intent = new Intent(getContext, classOf[AuthenticationActivity])
-      getContext.startActivity(intent)
+      getContext.startActivity(AuthenticationActivity.newIntent(getContext,true))
```
Comment (BradleyWire) [inline app/src/main/scala/com/waz/zclient/preferences/pages/ProfileView.scala]: formatting here too

File: app/src/main/res-v2/layout/activity_authentication.xml [hunk 3]
```diff
@@ -17,11 +17,36 @@
...
+        android:gravity="center"
+        android:text="@string/glyph__close"
+        android:textColor="@color/graphite"
+        android:textSize="@dimen/wire__text_size__regular"
+        android:visibility="gone"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <FrameLayout
+        android:id="@+id/layout_container"
```
Comment (BradleyWire) [inline app/src/main/res-v2/layout/activity_authentication.xml]: I think we can name this to something better

Comment (mejdoo) [reply app/src/main/res-v2/layout/activity_authentication.xml]: what do you suggest?

File: app/src/main/res-v2/layout/activity_authentication.xml [hunk 4]
```diff
@@ -17,11 +17,36 @@
...
+    <androidx.appcompat.widget.AppCompatTextView
+        android:id="@+id/authenticationCloseButton"
+        android:layout_width="@dimen/authentication_close_button_width"
+        android:layout_height="@dimen/authentication_close_button_height"
+        android:layout_gravity="top|end"
+        android:layout_marginTop="@dimen/authentication_close_button_margin"
+        android:layout_marginEnd="@dimen/authentication_close_button_margin"
+        android:fontFamily="@font/wire_glyphs"
+        android:gravity="center"
+        android:text="@string/glyph__close"
```
Comment (gizemb) [inline app/src/main/res-v2/layout/activity_authentication.xml]: why are we still using glyphs? isn't there an svg file in zeplin for this icon?

Comment (mejdoo) [reply app/src/main/res-v2/layout/activity_authentication.xml]: there is no icon in the android spec in Zeplin, just found that in the flow

Comment (gizemb) [reply app/src/main/res-v2/layout/activity_authentication.xml]: https://app.zeplin.io/styleguide/5de4f201d49c777852802121/components?coid=5de4f272fca2e1782d747534 here you can find

File: app/src/main/res-v2/layout/activity_authentication.xml (hunk 3 above)
Comment (BradleyWire) [reply app/src/main/res-v2/layout/activity_authentication.xml]: welcomeFragmentFrameLayoutContainer?

File: app/src/main/res-v2/layout/activity_authentication.xml (hunk 4 above)
Comment (mejdoo) [reply app/src/main/res-v2/layout/activity_authentication.xml]: thanks

File: app/src/main/res-v2/layout/activity_authentication.xml (hunk 3 above)
Comment (mejdoo) [reply app/src/main/res-v2/layout/activity_authentication.xml]: too long. I guess authenticationFrameLayoutContainer is fine

File: app/src/main/kotlin/com/waz/zclient/auth/AuthenticationActivity.kt [hunk 5]
```diff
@@ -5,17 +5,37 @@ import android.content.Intent
...
     }
 
     companion object {
         @JvmStatic
-        fun newIntent(context: Context) = Intent(context, AuthenticationActivity::class.java)
+        fun newIntent(context: Context, isCloseButtonVisible: Boolean): Intent =
+            Intent(context, AuthenticationActivity::class.java)
+                .putExtra(CLOSE_BUTTON_VISIBILITY_BUNDLE_KEY, isCloseButtonVisible)
+
+        private const val CLOSE_BUTTON_VISIBILITY_BUNDLE_KEY = "isCloseButtonVisible"
```
Comment (BradleyWire) [inline app/src/main/kotlin/com/waz/zclient/auth/AuthenticationActivity.kt]: Should constants live at the bottom of a companion object?

Comment (mejdoo) [reply app/src/main/kotlin/com/waz/zclient/auth/AuthenticationActivity.kt]: fixed thanks

**Comment by mejdoo** [comment]
We have done some refactoring and this PR is no longer valid.

**Pull request closed without merging**